packages/core/src/messagingProtocols/proteus/EventHandler/EventHandler.types.ts

```diff
@@ -30,7 +28,5 @@ export type EventHandlerParams = {
   dryRun?: boolean;
   event: BackendEvent;
   source: NotificationSource;
-  coreCryptoClient: CoreCrypto;
-  apiClient: APIClient;
-  useQualifiedIds: boolean;
+  decryptMessage: (encryptedPayload: Uint8Array, userId: QualifiedId, clientId: string) => Promise<Uint8Array>;
```

Comment:
I did an inversion of control here to avoid prop-drilling all the way down and keep the related work of encrypting/decrypting at the same location (since they are symetric, they should live together in the proteus service)

We just pass a `decrypt` function and the parent is responsible for passing this decrypt function